### DISCUSSION TITLE
Adding weight lifting data

### DISCUSSION THREAD

**Discussion by YUD4S** [tcgoetz/GarminDB#220] [Ideas]
Hello,

I see that there is not much weight-lifting data in the DB. It would be nice to get all exercises, sets, reps, volume, etc into the DB. 
So that we could visualize them with ease instead of the less-than-ideal visualization there is on Garmin Connect right now. 

Thanks,

**Comment by tcgoetz** [comment]
That isn't something I will be adding, but you are welcome to. If you do open a pull request against the develop branch with the new code.

**Reply by raphaelg** [reply]
In https://github.com/tcgoetz/GarminDB/wiki/Fit-Files : 

_General

The FIT file SDK can be found here: https://www.thisisant.com/resources/fit/
Unknown Messages

The following message types are unknown (not documented and I haven't figured them out):

(seen in monitoring FIT files)

    24
    211
    233

(seen in activity files)

    140
    141
    233

If you have any information about these message types, please pass it on._

The unknown messages are actually what is needed here :

| FIT Code | FIT SDK Message Name | Common File Context | Data Tracked |
| :--- | :--- | :--- | :--- |
| **24** | `monitoring` | Monitoring | Daily steps, floors, passive calories, active minutes |
| **140** | `spo2_data` | Activity / Monitoring | Blood oxygen saturation levels (%) |
| **141** | `hrv` | Activity | Beat-to-beat RR intervals (ms) |
| **211** | `monitoring_info` | Monitoring | Timezone offsets, monitoring cycle timestamps |
| **233** | `set` | Activity / Monitoring | Strength reps, weights, exercise types, and rest periods |
